From e94efec6aa0db31f9b72710df944bcb1d7833e14 Mon Sep 17 00:00:00 2001
From: Lei Chen <lei.chen@rock-chips.com>
Date: Thu, 16 Aug 2018 09:57:24 +0800
Subject: [PATCH] arm: dts: rk3128h-box-avb: add sdmmc_det for sdmmc pinctrl

This modification is to solve the problem that the SD card
is inserted after starting up, and the hot plug fails because
the det pin is initialized to GPIO

Change-Id: Id0d88259f7b30a75c07d7c2980563fc0d4148449
Signed-off-by: Lei Chen <lei.chen@rock-chips.com>
---
 arch/arm/boot/dts/rk3128h-box-avb.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/rk3128h-box-avb.dts b/arch/arm/boot/dts/rk3128h-box-avb.dts
index 97e4cddf0e6d..a480ba513417 100644
--- a/arch/arm/boot/dts/rk3128h-box-avb.dts
+++ b/arch/arm/boot/dts/rk3128h-box-avb.dts
@@ -52,6 +52,14 @@
 	};
 };
 
+&pinctrl {
+	sdmmc {
+		sdmmc_det: sdmmc-det {
+			rockchip,pins = <1 RK_PC1 RK_FUNC_1 &pcfg_pull_none>;
+		};
+	};
+};
+
 &secure_memory {
 	/*
 	 * enable like this:
@@ -59,3 +67,7 @@
 	 */
 	reg = <0x80000000 0x8000000>;
 };
+
+&sdmmc {
+	pinctrl-0 = <&sdmmc_pwr &sdmmc_clk &sdmmc_cmd &sdmmc_bus4 &sdmmc_det>;
+};
-- 
2.35.3

